src/java.base/share/classes/sun/security/x509/X509CRLImpl.java

```diff
@@ -280,13 +280,18 @@ public X500Principal getIssuerX500Principal() {
          *   prevCertIssuer if it does not exist
          */
         private X500Principal getCertIssuer(X509CRLEntryImpl entry,
-                                            X500Principal prevCertIssuer) {
+                X500Principal prevCertIssuer) throws CRLException {
 
             CertificateIssuerExtension ciExt =
                     entry.getCertificateIssuerExtension();
             if (ciExt != null) {
                 GeneralNames names = ciExt.getNames();
-                X500Name issuerDN = (X500Name) names.get(0).getName();
+                X500Name issuerDN;
+                try {
+                    issuerDN = (X500Name) names.get(0).getName();
```

Comment:
Use `instanceof X500Name` here to see if it is the right type instead of catching the `ClassCastException`.

Reply:
Fixed. Good Idea.